src/ucp/core/ucp_worker.h

```diff
@@ -253,6 +253,12 @@ struct ucp_worker_cm {
 };
 
 
+typedef struct ucp_worker_discard_req {
+    ucs_list_link_t req_list;
+    void            *req;
+} ucp_worker_discard_req_t;
```

Comment:
can we re-use some space in ucp_request to avoid this structure and extra allocation?

Reply:
done.